### PR TITLE
feat: add fedx-scripts serve

### PR DESCRIPTION
**Pull request by adamstankiewicz**
# Description

These changes propose adding a `serve` command to the `fedx-scripts` CLI such that consumers can create an NPM script to serve the production Webpack build from the generated `dist` directory when running `npm run build`. By serving the production build locally, engineers can have more confidence in their resulting production builds before it reaches a staging/production environment.

As is, there is no documented way to preview a production build of micro-frontends of the files in the `dist` directory. Simply opening the `index.html` does not work as the React routing breaks, amongst other issues.

The proposed `serve` command starts an [Express.js](https://expressjs.com/) serve running on the `PORT` setting defined in `env.config.js`. Note: the `.env*` files to configuration is now [deprecated](https://github.com/openedx/frontend-platform/blob/master/docs/decisions/0007-javascript-file-configuration.rst#decision) so the `serve` command is intentionally not supporting it.

[ADR](https://github.com/openedx/frontend-build/blob/deeac588e73733c7d9b9bfaf3b458ae7b473c303/docs/0003-fedx-scripts-serve.md)

[README documentation](https://github.com/openedx/frontend-build/blob/c756d237c9f5b9314aeaae01f39ac6b772f6808b/README.rst#serving-a-production-webpack-build-locally)

### CODE REVIEW

**Review by brobro10000 — APPROVED** [review]
Sad to see that the `.env*` isn't supported, but maybe the catalyst needed to start transitioning to `env.config.js` across other MFEs
👍🏽

File: bin/fedx-scripts.js [hunk 1]
```diff
@@ -1,4 +1,7 @@
 #!/usr/bin/env node
+
+const chalk = require('chalk');
```
Comment (davidjoy) [inline bin/fedx-scripts.js:3]: Love me some chalk. 😋

File: package.json [hunk 2]
```diff
@@ -55,23 +56,24 @@
     "eslint-plugin-jsx-a11y": "6.7.1",
     "eslint-plugin-react": "7.32.2",
     "eslint-plugin-react-hooks": "4.6.0",
+    "express": "^4.18.2",
```
Comment (davidjoy) [inline package.json:59]:
```suggestion
    "express": "4.18.2",
```

**Review by davidjoy — DISMISSED** [review]
Love it.  :shipit:

**Comment by adamstankiewicz** [comment]
@davidjoy I've updated this PR to add support for either `.env.development|private` file(s) _or_ `env.config.js` file. Please take another look when you get a chance!

File: package.json (hunk 2 above)
Comment (brobro10000) [inline package.json:59]: Bump on pinning this version 👍🏽

File: lib/scripts/serve.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+// Add process env vars. Currently used only for setting the
+// server port and the publicPath
+dotenv.config({
+  path: path.resolve(process.cwd(), '.env.development'),
+});
+
+// Allow private/local overrides of env vars from .env.development for config settings
+// that you'd like to persist locally during development, without the risk of checking
+// in temporary modifications to .env.development.
+resolvePrivateEnvConfig('.env.private');
```
Comment (brobro10000) [inline lib/scripts/serve.js:18]: 🥳 , but good stepping stone to transition over to `env.config.js`

**Review by brobro10000 — APPROVED** [review]
👍🏽

**Review by brobro10000 — APPROVED** [review]
Third times the charm!

**Pull request merged**